### ISSUE TITLE
WKWebView MediaTypesRequiringUserActionForPlayback

### ISSUE THREAD

**Issue by ALander26** [xamarin/xamarin-macios#3137]
### Steps to Reproduce

1. Setup application to load HTML5 audio page in WKWebView
2. WKWebView MediaTypesRequiringUserActionForPlayback configuration set for 'None' 
3. Run app to see if audio autostarts

One should note that the same code for iOS Swift works without a problem.

### Expected Behavior
The audio should play on page load.
### Actual Behavior
The audio does not play and requires user interaction to start the audio. Does not respect the MediaTypesRequiringUserActionForPlayback param.
### Environment

=== Visual Studio Community 2017 for Mac ===

Version 7.3 (build 799)
Installation UUID: a2d3a4a7-526c-464e-971f-b895e9683478
Runtime:
	Mono 5.4.1.7 (2017-06/e66d9abbb27) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)

	Package version: 504010007

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.0.0
	1.1.0
SDK: /usr/local/share/dotnet/sdk/2.0.0/Sdks
SDK Versions:
	2.0.0
	1.0.0-preview2-1-003177
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.4.1/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.0
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Apple Developer Tools ===

Xcode 9.2 (13772)
Build 9C40b

=== Xamarin.iOS ===

Version: 11.6.1.2 (Visual Studio Community)
Hash: 6857dfcc
Branch: xcode9.2
Build date: 2017-12-04 21:20:41-0500

=== Xamarin.Android ===

Version: 8.1.0.24 (Visual Studio Community)
Android SDK: /Users/perfectedtech/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.2    (API level 17)
		4.3    (API level 18)
		4.4    (API level 19)
		4.4.87 (API level 20)
		5.0    (API level 21)
		5.1    (API level 22)
		6.0    (API level 23)
		7.0    (API level 24)
		7.1    (API level 25)
		8.0    (API level 26)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 27.0.0
SDK Build Tools Version: 27.0.1

Java SDK: /usr
java version "1.8.0_152"
Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.152-b16, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Xamarin Inspector ===

Version: 1.3.2
Hash: 461f09a
Branch: 1.3-release
Build date: Tue, 03 Oct 2017 18:26:57 GMT
Client compatibility: 1

=== Xamarin.Mac ===

Version: 4.0.0.214 (Visual Studio Community)

=== Build Information ===

Release ID: 703000799
Git revision: df590420ea5bc681994baefff2eabdaad889fba9
Build date: 2017-12-05 09:23:45-05
Xamarin addins: 51068d6eaf3b6fb4fbc72b33cb869b0ebff04ec5
Build lane: monodevelop-lion-d15-5

=== Operating System ===

Mac OS X 10.13.0
Darwin 17.0.0 Darwin Kernel Version 17.0.0
    Thu Aug 24 21:48:19 PDT 2017
    root:xnu-4570.1.46~2/RELEASE_X86_64 x86_64

=== Enabled user installed addins ===

Internet of Things (IoT) development (Preview) 7.1

```
WebView = new WKWebView(instance.webview.Frame, new WKWebViewConfiguration()
                {
                    AllowsInlineMediaPlayback = true,
                    MediaTypesRequiringUserActionForPlayback = WKAudiovisualMediaTypes.None,
                    RequiresUserActionForMediaPlayback = false
                });
```

**Comment by VincentDondain** [comment]
Hi, I created a quick sample with your code and got the audio to play automatically (no interaction at all) on simulator and device.

Here's the sample: https://www.dropbox.com/s/418ywb8j98u34oy/Bug3137.zip?dl=0

I used the same Xamarin.iOS version as you: https://gist.github.com/VincentDondain/14ac1d1b1ee32881181cef99a238e1c7

Does my sample work for you? If not then you'll need to share some sample too so we can investigate this further.
